Sources/MapboxCoreNavigation/PassiveNavigationEventDetails.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import CoreLocation
+
+struct PassiveNavigationEventDetails: NavigationEventDetails {
+    let coordinate: CLLocationCoordinate2D?
+    let created = Date()
+    let sessionIdentifier: String
+    let startTimestamp: Date?
+    let driverMode = "freeDrive"
+    
+    var event: String?
+    var userIdentifier: String?
+    var feedbackType: FeedbackType?
+    var description: String?
+    var screenshot: String?
+    
+    var percentTimeInPortrait: Int = 0
+    var percentTimeInForeground: Int = 0
+    var totalTimeInForeground: TimeInterval = 0
+    var totalTimeInBackground: TimeInterval = 0
+    
+    init(dataSource: PassiveNavigationEventsManagerDataSource, sessionState: SessionState) {
+        coordinate = dataSource.rawLocation?.coordinate
+        sessionIdentifier = sessionState.identifier.uuidString
+        startTimestamp = sessionState.departureTimestamp
+        updateApplicationStatePercentages(sessionState)
+    }
+    
+    private enum CodingKeys: String, CodingKey {
```

Comment:
Is it possible to make this enum common for both structures? Or at least make sure that common cases have the same values?

Reply:
Unfortunately I don't see a way to do this. We can use only one enum for one Codable conformance and this cannot be the same enum since different events have different values. The raw value of enum cases can only be a string literal, this is a compiler restriction, so there is no way to have constants separately in one place.